financial-connections/src/main/java/com/stripe/android/financialconnections/features/common/ModalBottomSheetContent.kt

```diff
@@ -78,198 +95,213 @@ internal fun LegalDetailsBottomSheetContent(
         TextResource.Text(fromHtml(legalDetails.title))
     }
-    val learnMore = remember(legalDetails.learnMore) {
-        legalDetails.learnMore?.let { TextResource.Text(fromHtml(it)) }
+    val learnMore = remember(legalDetails.disclaimer) {
+        legalDetails.disclaimer?.let { TextResource.Text(fromHtml(it)) }
     }
-    val bullets = remember(legalDetails.body.bullets) {
-        legalDetails.body.bullets?.map { BulletUI.from(it) }
+    val links = remember(legalDetails.body.links) {
+        legalDetails.body.links.map { TextResource.Text(fromHtml(it.title)) }
     }
     ModalBottomSheetContent(
+        icon = legalDetails.icon,
         title = title,
         subtitle = null,
         onClickableTextClick = onClickableTextClick,
-        bullets = bullets,
         connectedAccountNotice = null,
         cta = legalDetails.cta,
-        learnMore = learnMore,
-        onConfirmModalClick = onConfirmModalClick,
-    )
+        disclaimer = learnMore,
+        onConfirmModalClick = onConfirmModalClick
+    ) {
+        links.forEachIndexed { index, link ->
+            item {
+                Divider(color = v3Colors.border, modifier = Modifier.padding(bottom = 16.dp))
+                AnnotatedText(
+                    modifier = Modifier.padding(bottom = 16.dp),
+                    text = link,
+                    defaultStyle = v3Typography.labelLargeEmphasized.copy(
+                        color = v3Colors.textBrand
+                    ),
+                    // remove annotation styles to avoid link underline (the default)
+                    annotationStyles = emptyMap(),
+                    onClickableTextClick = onClickableTextClick,
+                )
+                if (links.lastIndex == index) {
+                    Divider(color = v3Colors.border)
+                }
+            }
+        }
+    }
 }
 
 @Composable
 private fun ModalBottomSheetContent(
+    icon: Image?,
     title: TextResource,
     subtitle: TextResource?,
     onClickableTextClick: (String) -> Unit,
-    bullets: List<BulletUI>?,
     connectedAccountNotice: TextResource?,
     cta: String,
-    learnMore: TextResource?,
+    disclaimer: TextResource?,
     onConfirmModalClick: () -> Unit,
+    content: LazyListScope.() -> Unit,
 ) {
-    val scrollState = rememberScrollState()
-    Column {
-        Column(
-            Modifier
-                .weight(1f, fill = false)
-                .verticalScroll(scrollState)
-                .padding(24.dp)
-        ) {
-            AnnotatedText(
-                text = title,
-                defaultStyle = typography.heading.copy(
-                    color = colors.textPrimary
-                ),
-                annotationStyles = emptyMap(),
-                onClickableTextClick = onClickableTextClick
-            )
-            subtitle?.let {
-                Spacer(modifier = Modifier.size(4.dp))
+    Layout(
+        inModal = true,
+        content = {
+            item {
```

Comment:
Do you ever expect the sheet content to be larger than the screen height? If so, should this use `stickyHeader` as a precaution?

Reply:
Content could potentially be larger that the screen, specially on small devices. On these cases we want the title to also scroll with the content. That's the general approach from design across the flow, mainly because the button footer section is already sticky, and on large titles just a small portion of the screen becomes scrollable. 

This is how the modal looks like with content larger than the screen (modal opens full screen + all content scrolls):

https://github.com/stripe/stripe-android/assets/99293320/d745846b-43cc-413a-9327-74a8cba4499e

Reply:
That makes sense. Should the top padding in the bottom sheet be moved into the lazy column then, so things looks cleaner when scrolling? Right now, there’s a small gap between the scrolling content and the top of the sheet.

Reply:
replicated current behavior from [web](https://stripejsdevapp.corp.stripe.com/connections-auth-flow?profile=default&mode=test) to start with, but the `Layout` component gives us some scroll reaction capabilities. I'll play around and let you know!

Reply:
https://github.com/stripe/stripe-android/pull/7634/commits/469c927f043524f0e5e2c28de86a25764014ed7a